tidb-server/main.go

```diff
@@ -337,7 +352,7 @@ func overrideConfig() {
 	}
 }
 
-func validateConfig() {
+func validateConfig() error {
```

Comment:
I think It's better to print the error log then exit instead of return an error.

Reply:
Its really problematic to have exits anywhere other then the main function. The linters warn against this. In this case it is a private function in main.go, but still any function which exits is not testable.

I agree that being in an intermediate state where some return error and some still exit is not great, but I don't want to start cleaning up other code in this PR. Although it is still possible I might be forced to when I start writing test cases.

Reply:
OK, I see. But it still looks so strange. We need to clean up the other code anyway, but may not in this PR. It's better to add a TODO here.

Reply:
okay